### ISSUE TITLE
[MPLUGIN-290] Version 3.4 fails to parse enums with Regex patterns

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-plugin-tools#600]
**[Richard Birenheide](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=richardatsap)** opened **[MPLUGIN-290](https://issues.apache.org/jira/browse/MPLUGIN-290?redirect=false)** and commented

When having the file attached as Java source file, maven-plugin-plugin fails with the following error messages below.

Version 3.4 fails to parse the file, version 3.3 works fine.

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-plugin-plugin:3.4:descriptor (default-descriptor) on proje
ct extensionPointDocu-maven-plugin: Execution default-descriptor of goal org.apache.maven.plugins:maven-plugin-plugin:3.
4:descriptor failed: syntax error @[7,22] in file:/D:/D035816/repos/git-repos/fiori_extension_point_documenter/extension
PointDocu-maven-plugin/src/main/java/com/sap/suite/arch/test/Test.java -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-plugin-plu
gin:3.4:descriptor (default-descriptor) on project extensionPointDocu-maven-plugin: Execution default-descriptor of goal
 org.apache.maven.plugins:maven-plugin-plugin:3.4:descriptor failed: syntax error @[7,22] in file:/D:/D035816/repos/git-
repos/fiori_extension_point_documenter/extensionPointDocu-maven-plugin/src/main/java/com/sap/suite/arch/test/Test.java
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:225)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:84)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:59)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.singleThreadedBuild(LifecycleStarter.java:183)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:161)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:320)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:156)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:537)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:196)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:141)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:601)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:290)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:230)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:409)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:352)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-descriptor of goal org.apache.maven.plugi
ns:maven-plugin-plugin:3.4:descriptor failed: syntax error @[7,22] in file:/D:/D035816/repos/git-repos/fiori_extension_p
oint_documenter/extensionPointDocu-maven-plugin/src/main/java/com/sap/suite/arch/test/Test.java
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:110)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:209)
        ... 19 more
Caused by: com.thoughtworks.qdox.parser.ParseException: syntax error @[7,22] in file:/D:/D035816/repos/git-repos/fiori_e
xtension_point_documenter/extensionPointDocu-maven-plugin/src/main/java/com/sap/suite/arch/test/Test.java
        at com.thoughtworks.qdox.parser.impl.Parser.yyerror(Parser.java:1683)
        at com.thoughtworks.qdox.parser.impl.Parser.yyparse(Parser.java:1805)
        at com.thoughtworks.qdox.parser.impl.Parser.parse(Parser.java:1664)
        at com.thoughtworks.qdox.library.SourceLibrary.parse(SourceLibrary.java:210)
        at com.thoughtworks.qdox.library.SourceLibrary.parse(SourceLibrary.java:186)
        at com.thoughtworks.qdox.library.SourceLibrary.addSource(SourceLibrary.java:141)
        at com.thoughtworks.qdox.library.SortedClassLibraryBuilder.addSource(SortedClassLibraryBuilder.java:173)
        at com.thoughtworks.qdox.JavaProjectBuilder.addSource(JavaProjectBuilder.java:150)
        at com.thoughtworks.qdox.JavaProjectBuilder$2.visitFile(JavaProjectBuilder.java:222)
        at com.thoughtworks.qdox.directorywalker.DirectoryScanner.walk(DirectoryScanner.java:103)
        at com.thoughtworks.qdox.directorywalker.DirectoryScanner.walk(DirectoryScanner.java:91)
        at com.thoughtworks.qdox.directorywalker.DirectoryScanner.walk(DirectoryScanner.java:91)
        at com.thoughtworks.qdox.directorywalker.DirectoryScanner.walk(DirectoryScanner.java:91)
        at com.thoughtworks.qdox.directorywalker.DirectoryScanner.walk(DirectoryScanner.java:91)
        at com.thoughtworks.qdox.directorywalker.DirectoryScanner.walk(DirectoryScanner.java:91)
        at com.thoughtworks.qdox.directorywalker.DirectoryScanner.walk(DirectoryScanner.java:91)
        at com.thoughtworks.qdox.directorywalker.DirectoryScanner.scan(DirectoryScanner.java:81)
        at com.thoughtworks.qdox.JavaProjectBuilder.addSourceTree(JavaProjectBuilder.java:216)
        at com.thoughtworks.qdox.JavaProjectBuilder.addSourceTree(JavaProjectBuilder.java:203)
        at org.apache.maven.tools.plugin.extractor.annotations.JavaAnnotationsMojoDescriptorExtractor.discoverClasses(Ja
vaAnnotationsMojoDescriptorExtractor.java:460)
        at org.apache.maven.tools.plugin.extractor.annotations.JavaAnnotationsMojoDescriptorExtractor.discoverClasses(Ja
vaAnnotationsMojoDescriptorExtractor.java:434)
        at org.apache.maven.tools.plugin.extractor.annotations.JavaAnnotationsMojoDescriptorExtractor.discoverClasses(Ja
vaAnnotationsMojoDescriptorExtractor.java:413)
        at org.apache.maven.tools.plugin.extractor.annotations.JavaAnnotationsMojoDescriptorExtractor.scanJavadoc(JavaAn
notationsMojoDescriptorExtractor.java:187)
        at org.apache.maven.tools.plugin.extractor.annotations.JavaAnnotationsMojoDescriptorExtractor.execute(JavaAnnota
tionsMojoDescriptorExtractor.java:105)
        at org.apache.maven.tools.plugin.scanner.DefaultMojoScanner.populatePluginDescriptor(DefaultMojoScanner.java:96)

        at org.apache.maven.plugin.plugin.AbstractGeneratorMojo.execute(AbstractGeneratorMojo.java:245)
        at org.apache.maven.plugin.plugin.DescriptorGeneratorMojo.execute(DescriptorGeneratorMojo.java:90)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:101)
        ... 20 more
[ERROR]
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
```

---

**Affects:** 3.4

**Attachments:**
- [Test.java](https://issues.apache.org/jira/secure/attachment/12731437/Test.java) (_168 bytes_)

**Issue Links:**
- [MPLUGIN-314](https://issues.apache.org/jira/browse/MPLUGIN-314) invalid requirement role generated in plugin.xml

- [MPLUGIN-320](https://issues.apache.org/jira/browse/MPLUGIN-320) JavaJavadocMojoDescriptorExtractor fails with Java 8 lambdas.

0 votes, 7 watchers

**Comment by jira-importer** [comment]
**[Richard Birenheide](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=richardatsap)** commented

Java source which fails the plugin.

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

NEeds to be fixed at [QDox](https://github.com/paul-hammant/qdox)

**Comment by jira-importer** [comment]
**[Guillaume Boué](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=gboue)** commented

Fixed implictly during the upgrade of QDox to version 2.0-M5, done in MPLUGIN-314 and MPLUGIN-320.

**Comment by jira-importer** [comment]
**[Paul Vonnahme](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=pmv)** commented

This ticket has the fix version as 3.5.1, but I don't see that version in maven central?  http://search.maven.org/#search%7Cgav%7C1%7Cg%3A%22org.apache.maven.plugins%22%20AND%20a%3A%22maven-plugin-plugin%22

**Comment by jira-importer** [comment]
**[Paul Vonnahme](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=pmv)** commented

Nevermind - I see when I click on 3.5.1 it shows as 'Unreleased': https://issues.apache.org/jira/projects/MPLUGIN/versions/12338196

Is there a date set for the release?
